include/semaphore.h

```diff
@@ -0,0 +1,52 @@
+#include <condition_variable>
```

Comment:
Add header block (copyright + year + GPL + preprocessor one-time-include check and definds).  rwqueue's header should be a good example.

Reply:
Hah, just did that 2 minutes before you posted, thanks!